java/code/src/com/suse/manager/reactor/messaging/RegisterMinionEventMessageAction.java

```diff
@@ -127,128 +132,131 @@ public void execute(EventMessage msg) {
+            if (saltbootInitrd) {
+                // if we have the "saltboot_initrd" grain we want to re-deploy an image via saltboot,
+                LOG.info("Applying saltboot for minion " + minionId);
+                applySaltboot(registeredMinion);
             }
-        );
+        }
     }
 
-    private void reactivateSystem(String minionId, String machineId, Optional<User> creator,
-            ValueMap grains, Optional<ActivationKey> reactivationKey) {
-        // The machine id may have changed, but we know from the reactivation key
-        // which system should become this one
-        reactivationKey
-            .flatMap(rak -> rak.getServer().asMinionServer())
-            .ifPresent(minion -> {
-                minion.setMachineId(machineId);
-                minion.setMinionId(minionId);
-            });
+    private Optional<MinionServer> getRegisteredMinion(String machineId) {
+        return  MinionServerFactory.findByMachineId(machineId);
     }
 
     /**
-     * Check if a minion is already registered and update it in case so
-     * @param minionId the minion id
-     * @param machineId the machine id that we are trying to register
-     * @param creator the optional User that created the minion
-     * @param isSaltSSH true if a salt-ssh system is bootstrapped
-     * @param grains grains of the minion
-     * @return true if minion already registered, false otherwise
+     * Reactivate the system
+     * @param minionId minion id of the minion
+     * @param machineId machine_id of the minion
+     * @param managmentKeyLabel managment_key grain
      */
-    public boolean checkIfMinionAlreadyRegistered(String minionId,
-                                                  String machineId,
-                                                  Optional<User> creator,
-                                                  boolean isSaltSSH, ValueMap grains) {
-        Optional<MinionServer> optMinion = MinionServerFactory.findByMachineId(machineId);
-        if (optMinion.isPresent()) {
-            MinionServer registeredMinion = optMinion.get();
-            String oldMinionId = registeredMinion.getMinionId();
-
-            if (!minionId.equals(oldMinionId)) {
-                LOG.warn("Minion '" + oldMinionId + "' already registered, updating " +
-                        "profile to '" + minionId + "' [" + machineId + "]");
-                registeredMinion.setName(minionId);
-                registeredMinion.setMinionId(minionId);
-                ServerFactory.save(registeredMinion);
-                SystemManager.addHistoryEvent(registeredMinion, "Duplicate Machine ID", "Minion '" +
-                        oldMinionId + "' has been updated to '" + minionId + "'");
-
-                if (!minionId.equals(oldMinionId)) {
-                    SALT_SERVICE.deleteKey(oldMinionId);
-                }
+    private void reactivateSystem(String minionId, String machineId, Optional<String> managmentKeyLabel) {
```

Comment:
It seems like the meaning of this method has changed significantly: before it used to mean "set minion and machine ids according to the reactivation key, if we have one" - now it means "conclude the kickstart session, if any, check that we have a Activation Key, that it is actually a reactivation key and only in that case then set minion and machine ids". I think those are conceptually separate now and we should either keep the original name and meaning or change both, or rearrange code in some other way.

Reply:
You are right, I missed some part. I have changed it now.